component/async/component.go

```diff
@@ -6,9 +6,10 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
+
```

Comment:
empty line can be removed.

Reply:
I am under the impression that the gold standard for imports is:
golang imports
<empty line>
project imports
<empty line>
vendor imports

In fact that's how goland formats imports. Are we doing it differently? If so, why?

Reply:
(nitpicky comment, but since we started this conversation here go my 2¢).

I'm not sure how goland formats imports and if it does anything different, but the `goimports` tool is what should be used in all cases, for consistency whether you're working with Goland, Vim, or ed.

What we should do is a) remove all newlines from imports block b) run `goimports -w .` and let it do its thing.

The case with `goimports` is it only sorts and fixes [*consecutive import lines*](https://github.com/golang/tools/blob/d49c4edd7d9658159e8d95cd24fb84d69c1f4472/internal/imports/sortimports.go#L18). Whenever it finds a blank line, it thinks that you've decided on a custom grouping, and tries to order imports in that specific group only.

That's why if you try to run it now, you'll think that nothing will change and that the file is correctly formatted. The way that it formats the data is 'std imports' + '\n' + 'non-local imports'

Reply:
sorry just adding opinion here as looking over this PR  for context reasons @mantzas why aren't we separating local packages in the imports. This is standard among most projects and can be integrated into `goimports` easily using the `-local` flag.

Reply:
e.g. `goimports -local beatlabs/patron`

Reply:
the problem is that there is no standard as you mention. if you look at go project there is only a distinction between std and other packages.
we should move forward with the most common one which is the one explained.
Example we in dispatch have this rule, patron has also this rule... there is also a linter that we are using that enforces this in dispatch gofumpt which BTW just made it to gopls...

Reply:
check out https://github.com/mvdan/gofumpt and https://github.com/golang/tools/blob/master/gopls/doc/settings.md#gofumpt-bool

Reply:
Ok @mantzas I understand that if `patron` is currently not grouping by this convention then we should ignore. I'm just surprised it doesn't. Gofmt/gopls also has the option for a `local` config param https://github.com/golang/tools/blob/master/gopls/doc/settings.md#local-string

Reply:
If there was an official Go convention believe me we would use it. Now we are using the convention to separate the `std` from all others which is the most common one in a lot of projects out there. Everybody has their own convention and unfortunately Go did not have one for this AFAIK...
I hope that Go will eventually have a standard also for this, like it has for a lot of others, so we do not need to make such a big discussion about these kind of things.

Reply:
makes sense